[SYCLomatic] Update CodePin doc (#1962)
oneapi-src/SYCLomatic: docs/dev_guide/migration/debug-with-codepin.rst

Enable CodePin
--------------

Enable CodePin with the ``–enable-codepin`` option. The instrumented CUDA program will be put

> in the folder ``dpct_output_codepin_cuda``.

Reviewer: what about instrumented SYCL code?

    Result[3]: (2, 3, 4)
    */

> After building and executing ``dpct_output_codepin_sycl/example.dp.cpp`` and ``dpct_output_debug/example.cu``,

Reviewer: ```suggestion
After building ``dpct_output_codepin_sycl/example.dp.cpp`` and ``dpct_output_debug/example.cu`` and executing the binaries built out
```

> After building ``dpct_output_codepin_sycl/example.dp.cpp`` and ``dpct_output_debug/example.cu`` and executing the binaries built out

Reviewer: should be dpct_output_codepin_cuda/example.cu ?

> the following reports will be generated.

Reviewer: ```suggestion
the following execution log files will be generated.
```

                "Elapse Time(ms)": "0",
                "CheckPoint": {
                    "d_a": {
                        "Type": "Pointer",
                        "Data": [
                            {
                                "Type": "sycl::int3",
                                "Data": [
                                    {
                                        "x": {
                                            "Type": "int",
                                            "Data": [
                                                1
                                            ]
                                        }
                                    },
                                    {
                                        "y": {
                                            "Type": "int",
                                            "Data": [
                                                2
                                            ]
                                        }
                                    },
            ...

The report helps identify where the runtime behavior of the CUDA and the SYCL
programs start to diverge from one another.

> Analyze the Data Checkpoints

Reviewer: L 322 should be updated as "Most Time-consuming Kernel" is not info of  Data Checkpoints . 
how about Analysis the  CodePin result?

                "Total Device Memory": "16882663424",
                "Elapse Time(ms)": "0",
                "CheckPoint": {
                    "d_a": {
                        "Type": "Pointer",
                        "Data": [
                            {
                                "Type": "int3",
                                "Data": [
                                    {
                                        "x": {
                                            "Type": "int",
                                            "Data": [
                                                1
                                            ]
                                        }
                                    },
                                    {
                                        "y": {
                                            "Type": "int",
                                            "Data": [
                                                2
                                            ]
                                        }
                                    },
          ...

     - .. code-block::
           :linenos:

            [
                {

>                     "ID": "/home/yyergg/workspace/simple_test/test.cu:84:3:prolog",

Reviewer: ```suggestion
                    "ID": "test.cu:84:3:prolog",
```

                "CheckPoint": {
                    "d_a": {
                        "Type": "Pointer",
                        "Data": [
                            {
                                "Type": "sycl::int3",
                                "Data": [
                                    {
                                        "x": {
                                            "Type": "int",
                                            "Data": [
                                                1
                                            ]
                                        }
                                    },
                                    {
                                        "y": {
                                            "Type": "int",
                                            "Data": [
                                                2
                                            ]
                                        }
                                    },
            ...

The report helps identify where the runtime behavior of the CUDA and the SYCL
programs start to diverge from one another.

Analyze the Data Checkpoints
-------

> codepin-report.py is a tool consumes the execution log files from both CUDA and SYCL code and performs auto analysis.

Reviewer: ```suggestion
codepin-report.py (also can be triggered by dpct/c2s --codepin-report) is a functionality of the compatibility tool consumes the execution log files from both CUDA and SYCL code and performs auto analysis.
```

Reviewer 2: Missing a word: suggest adding the word 'that' to Zhiming's edit above: codepin-report.py (can also be triggered by dpct/c2s --codepin-report) is a functionality of the compatibility tool that consumes the execution log files from both CUDA and SYCL code and performs auto analysis.

> codepin-report.py can identify the in consistent data value and report the stats data of the execution log including data checkpoints.

Reviewer: ```suggestion
codepin-report.py can identify the inconsistent data value and report the stats data of the execution.
```

> codepin-report.py is a tool consumes the data point files from both CUDA and SYCL and performs auto analysis of the data checkpoints.

Reviewer: ```suggestion
codepin-report.py is a tool consumes the execution log file generated from both CUDA and SYCL code and performs auto analysis.
```

> codepin-report.py consumes the execution log files from both CUDA and SYCL code with the following command line.

Reviewer: ```suggestion
codepin-report.py consumes the execution log files generated from both CUDA and SYCL code with the following command line.
```

> codepin-report.py can identify the in consistent data value and report the stats data of the data checkpoints.

Reviewer: ```suggestion
codepin-report.py can identify the in consistent data value and report the stats data of the execution log including data checkpoints.
```

> ``codepin-report.py [-h] --instrumented-cuda-log <file path> --instrumented-sycl-log <file path>``

Reviewer: --floating-point-comparison-epsilon

> codepin-report.py consumes the data point files from both CUDA and SYCL with the following commandline.

Reviewer: ```suggestion
codepin-report.py consumes the execution log file generated from both CUDA and SYCL code with the following command line.
```

Following is an example of the analysis report.

.. code-block::

    CodePin Summary
    Totally APIs count, 2
    Consistently APIs count, 2

>     Most Time-consuming Kernel(CUDA), /home/yyergg/workspace/codepin_demo/example.cu:26:3:epilog, time:8.2316

Reviewer: ```suggestion
    Most Time-consuming Kernel(CUDA), example.cu:26:3:epilog, time:8.2316
```

>     Most Time-consuming Kernel(SYCL), /home/yyergg/workspace/codepin_demo/example.cu:26:3:epilog, time:10.2575

Reviewer: ```suggestion
    Most Time-consuming Kernel(SYCL), example.cu:26:3:epilog, time:10.2575
```
